perma_web/perma/tests/test_models.py

```diff
@@ -1459,178 +1458,154 @@ def test_registrar_most_active_org_this_year(registrar, organization_factory, li
     assert registrar.most_active_org_this_year() == o2
 
 
-# Fixtures
+#
+# Link limit for sponsored users
+#
 
-def complex_user_with_bonus_link(in_subfolder=False):
-    user = LinkUser(link_limit=2, bonus_links=0)
-    user.save()
-    user.organizations.add(Organization.objects.get(pk=1))
-    sponsorship = Sponsorship(registrar=Registrar.objects.get(pk=1), user=user, created_by=user)
-    sponsorship.save()
-    subfolder = Folder(parent=user.top_level_folders()[0], name='Subfolder')
-    subfolder.save()
-    bonus_link = Link(created_by=user, bonus_link=True)
-    bonus_link.save()
-    if in_subfolder:
-        bonus_link.move_to_folder_for_user(subfolder, user)
+def test_sponsored_links_not_counted_against_personal_total(sponsored_user, link_factory):
+    assert sponsored_user.get_links_remaining()[0] == 10
+    link = link_factory(creation_timestamp=timezone.now().replace(day=1), guid="AAAA-AAAA", created_by=sponsored_user)
+
+    assert sponsored_user.get_links_remaining()[0] == 9
+    link.move_to_folder_for_user(sponsored_user.sponsorships.first().folders.first(), sponsored_user)
+    assert sponsored_user.get_links_remaining()[0] == 10
+
+
+#
+# Link limit / subscription related tests for registrars
+#
+
+
+def test_renaming_registrar_renames_top_level_sponsored_folders(registrar, sponsorship_factory):
+    new_name = 'A New Name'
+    sponsorship_factory(registrar=registrar)
+    sponsored_folders = Folder.objects.filter(sponsored_by=registrar).prefetch_related('parent')
+    assert registrar.name != new_name and sponsored_folders
+    for folder in sponsored_folders:
+        if folder.parent.is_sponsored_root_folder:
+            assert folder.name == registrar.name
+        else:
+            assert folder.name != registrar.name
+    registrar.name = new_name
+    # You need to save if you use refresh_from_db
```

Comment:
Indeed! A person uses `refresh_from_db` only after calling save: the aim is to get a clean copy of the object so that you trust the results. For instance, if you literally typed:
```
registrar.foo = 'bar'
registrar.save()
```
here, and then did `assert registrar.foo == 'bar'`... it would pass, since you are just checking the very object that you created. But, if you were to do:
```
registrar.foo = 'bar'
registrar.save()
registrar.refresh_from_db()
```
and then `assert registrar.foo == 'bar'`, it would fail... because the `Registrar` model HAS no field called `foo`, so the call to save is a no-op, and so a fresh copy of that registrar does not have foo set to bar.

Does that make sense?